cpp/src/streaming/cudf/shuffler.cpp

```diff
@@ -0,0 +1,134 @@
+/**
+ * SPDX-FileCopyrightText: Copyright (c) 2025, NVIDIA CORPORATION & AFFILIATES.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+
+#include <memory>
+#include <numeric>
+
+#include <rmm/cuda_stream_view.hpp>
+
+#include <rapidsmpf/streaming/cudf/shuffler.hpp>
+
+namespace rapidsmpf::streaming::node {
+
+namespace {
+
+/**
+ * @brief RAII wrapper for a CUDA event.
+ *
+ * Creates a CUDA event on construction and destroys it on destruction.
+ */
+struct CudaEventRAII {
+    cudaEvent_t event{};
+
+    CudaEventRAII(unsigned flags = cudaEventDisableTiming) {
+        RAPIDSMPF_CUDA_TRY(cudaEventCreateWithFlags(&event, flags));
+    }
+
+    ~CudaEventRAII() {
+        if (event) {
+            cudaEventRecord(event);
```

Comment:
```suggestion
            cudaEventDestroy(event);
```
Also, does an empty (uninitialised) event compare to bool as false?

Reply:
Yes, because `cudaEvent_t` is a pointer but I have clean up `CudaEventRAII` a bit: 9ca8800d77e9ecaf80fd81a54d50d7fc778b4b3c